### PR TITLE
I add validation form

### PR DESCRIPTION
**Pull request by AhmedNAgy25**
i add validation form and prevent user to go out from outlayer but enable him to close it usinig close button and make another copy of function which make this and add validation function

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -11,6 +11,40 @@ const nav = document.querySelector('.nav');
 const tabsContent = document.querySelectorAll('.operations__content');
 const header = document.querySelector('.header');
 
+
+///////////////////////////////////////
+// validation to form
+function creationValidation() {
```
Comment (Fekry-Ahmed) [inline script.js:17]: start function names with a verb. 
 what about `validateCreationForm`?

File: script.js [hunk 2]
```diff
@@ -11,6 +11,40 @@ const nav = document.querySelector('.nav');
...
+  var firstName = document.getElementById("firstName").value;
+  var lastName = document.getElementById("lastName").value;
+  var email = document.getElementById("email").value;
+  var validationError = document.getElementById("invalidMessage");
+  validationError.classList.add('error');
+  var isValid = true;
+  var errorMessages = [];
```
Comment (Fekry-Ahmed) [inline script.js:24]: Don't use `var`, use const and let instead.

File: script.js [hunk 3]
```diff
@@ -11,6 +11,40 @@ const nav = document.querySelector('.nav');
...
+
+  if(firstName=="" || lastName=="" || email==""){
```
Comment (Fekry-Ahmed) [inline script.js:26]: use `===` instead of `==`

File: script.js [hunk 4]
```diff
@@ -19,15 +53,21 @@ const openModal = function (e) {
   modal.classList.remove('hidden');
   overlay.classList.remove('hidden');
 };
-
+/*i make ke w copy of this finto enable x button in line 70 */
 const closeModal = function () {
+  if(creationValidation()){
```
Comment (Fekry-Ahmed) [inline script.js:58]: it's annoying if you restrict the user form closing the modal while the form is not valid

File: script.js [hunk 5]
```diff
@@ -19,15 +53,21 @@ const openModal = function (e) {
...
+  modal.classList.add('hidden');
+  overlay.classList.add('hidden');
+  }
+};
+const closeModal_btn = function () {
```
Comment (Fekry-Ahmed) [inline script.js:63]: Why do we need two version of handlers to close the modal? please remove it

In general:
- use camelCase naming: `closeModalBtn`
- but anyway `closeModalBtn` is not a good name for event handler , may we use `handleCloseModal`  or `onCloseModalClick`

File: script.js [hunk 6]
```diff
@@ -19,15 +53,21 @@ const openModal = function (e) {
   modal.classList.remove('hidden');
   overlay.classList.remove('hidden');
 };
-
+/*i make ke w copy of this finto enable x button in line 70 */
```
Comment (Fekry-Ahmed) [inline script.js:56]: remove comment please.

**Review by Fekry-Ahmed — CHANGES_REQUESTED** [review]

File: script.js (hunk 1 above)
Comment (AhmedNAgy25) [reply script.js:17]: ok

File: script.js (hunk 3 above)
Comment (AhmedNAgy25) [reply script.js:26]: why?

File: script.js (hunk 4 above)
Comment (AhmedNAgy25) [reply script.js:58]: i let him close but using x on top right

File: script.js (hunk 5 above)
Comment (AhmedNAgy25) [reply script.js:63]: to make one work depend on validation and one work normally

File: script.js (hunk 2 above)
Comment (AhmedNAgy25) [reply script.js:24]: ok

File: script.js (hunk 3 above)
Comment (Fekry-Ahmed) [reply script.js:26]: check this
https://blog.webdevsimplified.com/2020-08/==-vs-===/

Comment (AhmedNAgy25) [reply script.js:26]: understood

File: script.js (hunk 4 above)
Comment (AhmedNAgy25) [reply script.js:58]: i deal with this

File: script.js (hunk 5 above)
Comment (AhmedNAgy25) [reply script.js:63]: after some edit i make this with one

File: script.js (hunk 6 above)
Comment (AhmedNAgy25) [reply script.js:56]: ok

**Review by Fekry-Ahmed — APPROVED** [review]

**Pull request merged**